### PR TITLE
Support older versions of `rpy2`>=3.3.

### PR DESCRIPTION
**Pull request by matthewwardrop**
Hi @lgautier ,

I have unfortunately been having many issues with the `rpy2` 3.5.x release series, and haven't had much success getting any of these versions working reliably in our stack. I did still want to migrate to `rpy2-arrow` from our custom implementation, since our version (though slightly faster) requires compilation against specific versions of arrow (which makes distribution annoying). As such, please accept this patch to add support for older versions of `rpy2` (we have been having no issues with 3.4.5). I've limited this support arbitrarily to `rpy2` >=3.3 since I doubt anyone will want to use anything older, and did want to provide some lower-bound.

(Note: The current bound in pyproject.toml is >=3.5, but in actual fact you currently require >=3.5.2 due to the threading changes).

### CODE REVIEW

**Comment by lgautier** [comment]
This makes me realize that the test matrix only looks at the latest rpy2 release. I should probably add rpy2 version numbers. Later.

**Review by lgautier — APPROVED** [review]
LGTM

File: pyproject.toml
Comment (matthewwardrop) [inline pyproject.toml:29]: Why are you excluding rpy2 3.4.x? (that's what we were originally hoping to use).

Were you attempting to do something like: `rpy2>=3.3; python_version > "3.4"`?

Comment (lgautier) [reply pyproject.toml:29]: Ah. I had understood your first comment in the PR as "I need 3.3 but can't test versions between that and latest (3.5)".

Comment (lgautier) [reply pyproject.toml:29]: Let me revert then.

Comment (matthewwardrop) [reply pyproject.toml]: Ah... thank you! I have tested somewhat thoroughly this with rpy2 3.4.x . I'm happy for the lower constraint to be 3.4 if you like. I just chose 3.3 because that goes far enough back (to pandas 1.0.0 days) that I doubt it makes sense to go further. You may know of some reason to exclude 3.3.x.

**Comment by matthewwardrop** [comment]
You would have to merge this, btw. I don't have merge rights :).

**Comment by matthewwardrop** [comment]
@lgautier I should have mentioned earlier, but this is blocking some internal work as I have already migrated some code to sit on top of this, but due to the rpy2 3.5.x bugs (for which I've submitted patches into `rpy2`), I cannot deploy things until these patches are merged and the conda package updated. If you are deep in other things, no pressure; but the sooner things can be released the better for me (otherwise I'll have to release some internal packages that shadow this package, which would be a wee bit annoying).

**Comment by lgautier** [comment]
Got it. Is ≥ 3.4 good enough / unblocking you?

**Comment by matthewwardrop** [comment]
Yes!

**Review by lgautier — APPROVED** [review]

**Comment by lgautier** [comment]
I will release and upload to pypi in an hour or so.

**Comment by matthewwardrop** [comment]
Thank you! 🙏

**Pull request merged**
